featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -460,8 +460,37 @@ def test_abides_by_max_depth_param(es):
 
         features = dfs_obj.build_features()
         for f in features:
-            # last feature is identity feature which doesn't count
-            assert (f.get_depth() <= i + 1)
+            assert (f.get_depth() <= i)
+
+
+def test_max_depth_single_table(transform_es):
+    assert len(transform_es.entity_dict) == 1
+    for i in [-1, 0, 1, 2, None]:
+        def make_dfs_obj(max_depth=i):
+            dfs_obj = DeepFeatureSynthesis(target_entity_id='first',
+                                           entityset=transform_es,
+                                           trans_primitives=[AddNumeric],
+                                           max_depth=max_depth)
+            return dfs_obj
```

Comment:
Doesn't really impact things, but should this be moved outside of the for loop and the default value for `max_depth` be removed since we are always calling it below with a value defined?